zio-http/src/main/scala/zhttp/service/Handler.scala

```diff
@@ -53,23 +54,31 @@ private[zhttp] final case class Handler[R](
         val hasBody = canHaveBody(jReq)
         log.debug(s"HasBody: [${hasBody}]")
         log.debug(s"HttpRequest: [${jReq.method} ${jReq.uri()}]")
-        if (hasBody) ctx.channel().config().setAutoRead(false): Unit
+        if (hasBody) {
+          ctx.channel().attr(bodyReadFlag).set(false)
```

Comment:
This will introduce a bug when the same channel is being used for multiple requests.